<p>A wrong scope may cause problems to import the API definition into WSO2 or allow all users to call the endpoint.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      key: read
      roles: READ_ROLE
      description: Allows users to view records
paths:
  /pets:
    get:
      x-scope: view
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      key: read
      roles: READ_ROLE
      description: Allows users to view records
paths:
  /pets:
    get:
      x-scope: read
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      key: read
      roles: READ_ROLE
      description: Allows users to view records
paths:
  /pets:
    get:
      x-scope: view
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      key: read
      roles: READ_ROLE
      description: Allows users to view records
paths:
  /pets:
    get:
      x-scope: read
</pre>

