<p>A description can help other developers to understand the correct use of the scope.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read
      description: Allows users to read data
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
x-wso2-security:
  apim:
    x-wso2-scopes:
    - name: read
      roles: READ_ROLE
      key: read
      description: Allows users to read data
</pre>
