<p>A response not compliant with the standard may cause application issues.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /endpoint:
      get:
        responses:
          200:
            description: Ok
            schema:
              $ref: '#/definitions/successResponse'
          204:
            description: No content
          400:
            description: Error
            schema:
              $ref: '#/definitions/errorResponse'
          default:
            description: Unknown
  
  definitions:
    successResponse:
      type: object
      properties:
        payload:
          type: object
          properties:
            tipos:
              type: array
              items:
                type: string
    errorResponse:
      type: object
      properties:
        error:
          type: object
          properties:
            code:
              type: integer <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'code' must be of type string}}</span>
            message:
              type: integer <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'message' must be of type string}}</span>
            details:
              type: integer <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'details' must be of type array}}</span>
            httpStatus:
              type: string <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'httpStatus' must be of type integer}}</span>
          required:
            - code
            - message
            - httpStatus
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
          schema:
            $ref: '#/definitions/response'
        204:
          description: No content
        400:
          description: Error
          schema:
            $ref: '#/definitions/response'
        default:
          description: Unknown
definitions:
  response:
    type: object
    allOf:
      - $ref: '#/definitions/standard_response'
    properties:
      data:
        type: object
        properties:
          tipos:
            type: array
            items:
              type: string
    required:
      - data
      - status
  standard_response:
    type: object
    properties:
      status:
        type: object
        properties:
          http_status:
            type: string
          code:
            type: integer
          errors:
            type: array
            items:
              type: object
              properties: 
                name:
                  type: string
                value:
                  type: string
          description:
            type: string
          internal_code:
            type: string
        required:
          - http_status
          - code
          - description  
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: "3.0.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /endpoint:
      get:
        responses:
          200:
            description: Ok
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/successResponse'
          204:
            description: No content
          400:
            description: Error
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/errorResponse'
          default:
            description: Unknown
  
  components:
    schemas:
      successResponse:
        type: object
        properties:
          payload:
            type: object
            properties:
              tipos:
                type: array
                items:
                  type: string
      errorResponse:
        type: object
        properties:
          error:
            type: object
            properties:
              code:
                type: integer <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'code' must be of type string}}</span>
              message:
                type: integer <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'message' must be of type string}}</span>
              details:
                type: integer <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'details' must be of type array}}</span>
              httpStatus:
                type: string <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR029: 'httpStatus' must be of type integer}}</span>
            required:
              - code
              - message
              - httpStatus
  
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        200:
          description: Ok
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        204:
          description: No content
        400:
          description: Error
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        default:
          description: Unknown
components:
  schemas:
    response:
      type: object
      allOf:
        - $ref: '#/components/schemas/standard_response'
      properties:
        data:
          type: object
          properties:
            tipos:
              type: array
              items:
                type: string
      required:
        - data
        - status
    standard_response:
      type: object
      properties:
        status:
          type: object
          properties:
            http_status:
              type: string
            code:
              type: integer
            errors:
              type: array
              items:
                type: object
                properties: 
                  name:
                    type: string
                  value:
                    type: string
            description:
              type: string
            internal_code:
              type: string
          required:
            - http_status
            - code
            - description
            - errors  
</pre>