<p>A paged response not compliant with the standard may cause application issues.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        206:
          description: Ok
        204:
          description: No content
        400:
          description: Error
        default:
          description: Unknown
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        206:
          description: Ok
          schema:
            $ref: '#/definitions/response'
        204:
          description: No content
        400:
          description: Error
          schema:
            $ref: '#/definitions/response'
        default:
          description: Unknown

definitions:
  response:
    type: object
    allOf:
      - $ref: '#/definitions/paged_response'
    properties:
      payload:
        type: object
        properties:
          values:
            type: array
            items:
              type: string
  paged_response:
    type: object
    properties:
      paging:
        type: object
        properties:
          init:
            type: integer
          limit:
            type: integer
          total:
            type: integer
          numPages:
            type: integer
          links:
            type: object
            properties:
              self:
                $ref: '#/definitions/link'
              first:
                $ref: '#/definitions/link'
              previous:
                $ref: '#/definitions/link'
              next:
                $ref: '#/definitions/link'
              last:
                $ref: '#/definitions/link'
            required:
              - self
              - previous
              - next
        required:
          - init
          - limit
          - links
  link:
    type: object
    properties:
      href:
        type: string
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        206:
          description: Ok
        204:
          description: No content
        400:
          description: Error
        default:
          description: Unknown
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /endpoint:
    get:
      responses:
        206:
          description: Ok
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        204:
          description: No content
        400:
          description: Error
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/response'
        default:
          description: Unknown

components:
  schemas:
    response:
      type: object
      allOf:
        - $ref: '#/components/schemas/paged_response'
      properties:
        payload:
          type: object
          properties:
            values:
              type: array
              items:
                type: string
    paged_response:
      type: object
      properties:
        paging:
          type: object
          properties:
            init:
              type: integer
            limit:
              type: integer
            total:
              type: integer
            numPages:
              type: integer
            links:
              type: object
              properties:
                self:
                  $ref: '#/components/schemas/link'
                first:
                  $ref: '#/components/schemas/link'
                previous:
                  $ref: '#/components/schemas/link'
                next:
                  $ref: '#/components/schemas/link'
                last:
                  $ref: '#/components/schemas/link'
              required:
                - self
                - previous
                - next
          required:
            - init
            - limit
            - links
    link:
      type: object
      properties:
        href:
          type: string
</pre>