<p>Request body must contain only references ($ref)</p>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.0
  info:
    title: Sample API
    version: '1.0.0'
  paths:
    /pets:
      post:
        summary: Create a pet
        requestBody: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR092: Request body must contain only references ($ref)}}</span>
          description: The pet to create
          content:
            application/json:
              schema:
                type: object
                properties:
                  id:
                    type: integer
                  name:
                    type: string
        responses:
          '201':
            description: Pet created successfully
            content:
              application/json:
                schema:
                  type: object
                  properties:
                    id:
                      type: integer
                    name:
                      type: string
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.0
  info:
    title: Sample API
    version: '1.0.0'
  paths:
    /pets:
      post:
        summary: Create a pet
        requestBody:
          $ref: '#/components/requestBodies/PetBody'
        responses:
          '201':
            description: Pet created successfully
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/Pet'
  components:
    requestBodies:
      PetBody:
        description: The pet to create
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Pet'
    schemas:
      Pet:
        type: object
        properties:
          id:
            type: integer
          name:
            type: string
</pre>