<p>When reading a URL, it must be possible to interpret which resources it is referring to in a concrete way</p>
<p>Ambiguous names in resources must be avoided. The recommendation is to use domain specific names.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    version: 1.0.0
    title: Swagger Petstore
  paths:
    /elements: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR032: Ambiguous path parts not encouraged: elements}}</span>
      get:
        responses:
          200:
            description: Ok
    /a/nested/items: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR032: Ambiguous path parts not encouraged: items}}</span>
      get:
        responses:
          200:
            description: Ok
    /another/{param}/with/valores: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR032: Ambiguous path parts not encouraged: valores}}</span>
      get:
        responses:
          200:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices/{id}/products:
    get:
      parameters:
        - name: id
          in: path
          required: true
          type: integer
      responses:
        200:
          description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices/{id}/elements: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR032: Ambiguous path parts not encouraged: elements}}</span>
    get:
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices/{id}/products:
    get:
      parameters:
      - name: id
        in: path
        required: true
        schema:
          type: integer
      responses:
        200:
          description: Ok
</pre>