<p>in the verb Put, some path structures are not encouraged. This can be a hint of a design flaw.</p>
<h2>Forbidden paths by default</h2>
<table>
  <thead>
    <tr>
      <th>VERB</th>
      <th>STRUCTURE</th>
    </tr>
  </thead>
  <tbody>
    <tr>
        <td>PUT</td>
        <td>/resources/get</td>
      </tr>
      <tr>
        <td>PUT</td>
        <td>/resources/delete</td>
      </tr>
  </tbody>
</table>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /resources/get:
    put: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR105: Operation not recommended for resource path: PUT /resources/get}}</span>
      responses:
        '200':
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
    swagger: "2.0"
    info:
      title: Swagger Petstore
      version: 1.0.0
    paths:
      /resources/{r_id}:
        put: 
          responses:
            '200':
              description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/get:
    put: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR105: Operation not recommended for resource path: PUT /resources/get}}</span>
      responses:
        '200':
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
servers:
- url: /
paths:
  /resources/me: 
    put: 
      responses:
        '200':
          description: Ok
</pre>
