<p>All parameters in query must be defined as optional</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    title: Swagger Petstore
    version: 1.0.0
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $expand
            type: array
            required: true <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR060: All parameters in query must be defined as optional}}</span>
            items:
              type: string
        responses:
          200:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $expand
          type: array
          required: false
          items:
            type: string
      responses:
        200:
          description: Ok
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.1
  info:
    title: Swagger Petstore
    version: 1.0.0
  paths:
    /pets:
      get:
        parameters:
          - in: query
            name: $orderby
            required: true <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR060: All parameters in query must be defined as optional}}</span>
            schema:
              type: array
              items:
                type: string
        responses:
          200:
            description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /pets:
    get:
      parameters:
        - in: query
          name: $orderby
          required: false
          schema:
            type: array
            items:
              type: string
      responses:
        200:
          description: Ok
</pre>