<p>It is recommended to use examples instead of example as some tools like microcks use this section of the definition</p>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.0
  info:
    title: API con Example
    version: 1.0.0
  paths:
    /pets:
      get:
        summary: Devuelve una lista de mascotas
        responses:
          '200':
            description: Lista exitosa de mascotas
            content:
              application/json:
                schema:
                  type: array
                  items:
                    $ref: '#/components/schemas/Pet'
                example: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR094: It is recommended to use examples instead of example as some tools like microcks use this section of the definition}}</span>
                  - name: Whiskers
                    animalType: Cat
                  - name: Daisy
                    animalType: Dog
  components:
    schemas:
      Pet:
        type: object
        properties:
          name:
            type: string
          animalType:
            type: string
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
  openapi: 3.0.0
  info:
    title: API con Examples
    version: 1.0.0
  paths:
    /pets:
      get:
        summary: Devuelve una lista de mascotas
        responses:
          '200':
            description: Lista exitosa de mascotas
            content:
              application/json:
                schema:
                  type: array
                  items:
                    $ref: '#/components/schemas/Pet'
                examples:
                  catExample:
                    value:
                      - name: Whiskers
                        animalType: Cat
                  dogExample:
                    value:
                      - name: Daisy
                        animalType: Dog
  components:
    schemas:
      Pet:
        type: object
        properties:
          name:
            type: string
          animalType:
            type: string
</pre> 