<p>A string type with invalid format or without format may cause developers to use the wrong variable types in the API implementation.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          schema:
            type: array
            items:
              type: object
              properties:
                date:
                  type: string <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR037: String types requires a valid format}}</span>
                  format: 'dd/mm/yyyy'
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          schema:
            type: array
            items:
              type: object
              properties:
                date:
                  type: string
                  format: date
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          content:
            application/json:
              schema:
                type: array
                items:
                  type: object
                  properties:
                    date:
                      type: string <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR037: String types requires a valid format}}</span>
                      format: dd/mm/yyyy
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: 1.0.0
paths:
  /invoices:
    get:
      responses:
        200:
          description: Invoices.
          content:
            application/json:
              schema:
                type: array
                items:
                  type: object
                  properties:
                    date:
                      type: string
                      format: date
</pre>