<p>When defining an API, the request data formats must be defined.</p>
<p>For <b>OpenAPI 2</b> specify at least a MIME Type understood by the API in the "consumes" keyword is mandatory.</p>
<p>For <b>OpenAPI 3</b> specify at least a Media Type in the content of the response body is mandatory.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    put: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR007: Section produces is mandatory}}</span>
      responses: 
        200:
          description: OK
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
produces:
  - application/json
paths:
  /pets:
    put:
      responses: 
        200:
          description: OK
</pre>
<p>Or</p>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    put:
      produces:
        - application/json
      responses: 
        200:
          description: OK
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR007: Section produces is mandatory}}</span>
      responses:
        200:
          description: Ok
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.0
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets:
    post:
      responses:
        200:
          description: Ok
          content:
            application/json:
              schema:
                type: object
</pre>