<p>Descriptions must begin with a capital letter, end with a period and not be empty</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
    swagger: '2.0'
    info:
      title: Sample API
      version: 1.0.0
      description: esta es la descripci贸n de la API. <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR086: All descriptions must begin with a capital letter and end with a period}}</span>
    
    paths:
      /pets:
        get:
          summary: Lista de mascotas 
          description: esta ruta devuelve una lista de mascotas. <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR086: All descriptions must begin with a capital letter and end with a period}}</span>
          responses:
            '200':
              description: Una lista de mascotas.
              schema:
                $ref: '#/definitions/Pet'
    
    definitions:
      Pet:
        type: object
        properties:
          id:
            type: integer
            format: int64
          name:
            type: string
        description: Representa una mascota <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR086: All descriptions must begin with a capital letter and end with a period}}</span>
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
    swagger: '2.0'
    info:
      title: Sample API
      version: 1.0.0
      description: Esta es la descripci贸n de la API.
    
    paths:
      /pets:
        get:
          summary: Lista de mascotas.
          description: Esta ruta devuelve una lista de mascotas.
          responses:
            '200':
              description: Una lista de mascotas.
              schema:
                $ref: '#/definitions/Pet'
    
    definitions:
      Pet:
        type: object
        properties:
          id:
            type: integer
            format: int64
          name:
            type: string
        description: Representa una mascota.
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
    openapi: 3.0.0
    info:
    title: Sample API
    version: 1.0.0
    description: Esta es la descripci贸n de la API <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR086: All descriptions must begin with a capital letter and end with a period}}</span>

    paths:
    /pets:
        get:
        summary: lista de mascotas. 
        description: esta ruta devuelve una lista de mascotas <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR086: All descriptions must begin with a capital letter and end with a period}}</span>
        responses:
            '200':
            description: Una lista de mascotas.

    components:
    schemas:
        Pet:
        type: object
        properties:
            id:
            type: integer
            format: int64
            name:
            type: string
        description: Representa una mascota <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR086: All descriptions must begin with a capital letter and end with a period}}</span>
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
    openapi: 3.0.0
    info:
      title: Sample API
      version: 1.0.0
      description: Esta es la descripci贸n de la API.
    
    paths:
      /pets:
        get:
          summary: Lista de mascotas.
          description: Esta ruta devuelve una lista de mascotas.
          responses:
            '200':
              description: Una lista de mascotas.
    
    components:
      schemas:
        Pet:
          type: object
          properties:
            id:
              type: integer
              format: int64
            name:
              type: string
          description: Representa una mascota.
</pre>