<p>The convention for naming properties in a JSON is "camelCase" (all words together and capitalizing the first letter of each word after the first), therefore, the path parameters, the query parameters, request body  properties and response body properties must be in this convention.</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
  swagger: "2.0"
  info:
    title: Swagger Petstore
    version: "1.0"
  paths:
    /pets/{pet-id}:
      patch:
        consumes:
          - application/json
        produces:
          - application/json
        parameters:
          - in: path
            name: pet-id
            type: integer
            required: true
          - in: query
            name: pet_Type
            type: string
          - in: body
            name: pet
            description: Pet
            schema:
              type: object
              required:
                - PET_NAME
              properties:
                PET_NAME:
                  type: string
                photoUrls:
                  type: array
                  items:
                    type: string
        responses:
          200:
            description: OK
            schema:
              type: object
              properties:
                petId:
                  type: integer
                petType:
                  type: string
                petName:
                  type: string
                photoUrls:
                  type: array
                  items:
                    type: string
  </pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets/{petId}:
    patch:
      consumes:
        - application/json
      produces:
        - application/json
      parameters:
        - in: path
          name: petId
          type: integer
          required: true
        - in: query
          name: petType
          type: string
        - in: body
          name: pet
          description: Pet
          schema:
            type: object
            required:
              - petName
            properties:
              petName:
                type: string
              photoUrls:
                type: array
                items:
                  type: string
      responses:
        200:
          description: OK
          schema:
            type: object
            properties:
              petId:
                type: integer
              petType:
                type: string
              petName:
                type: string
              photoUrls:
                type: array
                items:
                  type: string
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets/{pet-id}:
    patch:
      parameters:
      - name: pet-id
        in: path
        required: true
        schema:
          type: integer
      - name: PET_TYPE
        in: query
        schema:
          type: string
      requestBody:
        description: Pet
        content:
          application/json:
            schema:
              required:
              - pet_name
              type: object
              properties:
                pet_name:
                  type: string
                photoUrls:
                  type: array
                  items:
                    type: string
        required: true
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  pet-id:
                    type: integer
                  PET_TYPE:
                    type: string
                  petName:
                    type: string
                  photoUrls:
                    type: array
                    items:
                      type: string
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: 3.0.1
info:
  title: Swagger Petstore
  version: "1.0"
paths:
  /pets/{petId}:
    patch:
      parameters:
      - name: petId
        in: path
        required: true
        schema:
          type: integer
      - name: petType
        in: query
        schema:
          type: string
      requestBody:
        description: Pet
        content:
          application/json:
            schema:
              required:
              - petName
              type: object
              properties:
                petName:
                  type: string
                photoUrls:
                  type: array
                  items:
                    type: string
        required: true
      responses:
        200:
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  petId:
                    type: integer
                  petType:
                    type: string
                  petName:
                    type: string
                  photoUrls:
                    type: array
                    items:
                      type: string
</pre>