<p>The last part of the path should be the API version</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
  version: 1.0.0
  title: Swagger Petstore
basePath: /api-seguros/v1.0 <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR100: Last path part must be the API version, indicated with the prefix 'v' and the version number as integer}}</span>
paths: {}
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
swagger: "2.0"
info:
    version: 1.0.0
    title: Swagger Petstore
basePath: /api-seguros/v1
paths: {}
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: https://petstore.swagger.io/api-seguros/v1.0 <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR100: Last path part must be the API version, indicated with the prefix 'v' and the version number as integer}}</span>
paths: {}
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
openapi: "3.0.0"
info:
    version: 1.0.0
    title: Swagger Petstore
servers:
    - url: https://petstore.swagger.io/api-seguros/v1
paths: {}
</pre>