<p>License information cannot be empty</p>
<h2>Noncompliant Code Example (OpenAPI 2)</h2>
<pre>
    swagger: "2.0"
    info:
      title: API de Ejemplo
      version: "1.0.0"
      description: Esta es una API de ejemplo.
      termsOfService: "http://ejemplo.com/terminos/"
      contact:
        name: Soporte de API
        url: "http://ejemplo.com/contacto"
        email: "contacto@ejemplo.com"
      license: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR110: License information cannot be empty}}</span>
    host: "api.ejemplo.com"
    schemes:
      - https
    paths: {}
</pre>
<h2>Compliant Solution (OpenAPI 2)</h2>
<pre>
    swagger: "2.0"
    info:
      title: API de Ejemplo
      version: "1.0.0"
      description: Esta es una API de ejemplo.
      termsOfService: "http://ejemplo.com/terminos/"
      contact:
        name: Soporte de API
        url: "http://ejemplo.com/contacto"
        email: "contacto@ejemplo.com"
      license:
        name: "Apache 2.0"
        url: "https://www.apache.org/licenses/LICENSE-2.0.html"
    host: "api.ejemplo.com"
    schemes:
      - https
    paths: {}
</pre>
<h2>Noncompliant Code Example (OpenAPI 3)</h2>
<pre>
    openapi: 3.0.0
    info:
      title: API de Ejemplo
      version: 1.0.0
      description: Esta es una API de ejemplo.
      termsOfService: http://ejemplo.com/terminos/
      contact:
        name: Soporte de API
        url: http://ejemplo.com/contacto
        email: contacto@ejemplo.com
      license: <span class="error-info" style="color: #FD8E18;"># Noncompliant {{OAR110: License information cannot be empty}}</span>
    paths: {}
</pre>
<h2>Compliant Solution (OpenAPI 3)</h2>
<pre>
    openapi: 3.0.0
    info:
      title: API de Ejemplo
      version: 1.0.0
      description: Esta es una API de ejemplo.
      termsOfService: http://ejemplo.com/terminos/
      contact:
        name: Soporte de API
        url: http://ejemplo.com/contacto
        email: contacto@ejemplo.com
      license:
        name: Apache 2.0
        url: https://www.apache.org/licenses/LICENSE-2.0.html
    paths: {}
</pre>